src/Phinx/Db/Adapter/MysqlAdapter.php

```diff
@@ -238,7 +238,7 @@ public function createTable(Table $table, array $columns = [], array $indexes =
                    ->setIdentity(true);
 
             array_unshift($columns, $column);
-            if (isset($options['primary_key'])) {
+            if (isset($options['primary_key']) && $options['id'] !== $options['primary_key'] && [$options['id']] !== $options['primary_key']) {
```

Comment:
Please use the recommend cast I gave you.
This way you force a double array nesting in some cases.

Reply:
Wouldnt then

    (array)$options['id'] !== (array)$options['primary_key']

be enough?

Reply:
Yup. Completely spaced on how casts work. 🤦‍♂

Fixed properly and more succinctly.